Dockerfile.build

```diff
@@ -1,101 +1,91 @@
 ARG GOLANG_VERSION=1.21.3
 ARG CMAKE_VERSION=3.22.1
 ARG CUDA_VERSION=11.3.1
-ARG ROCM_VERSION=5.7.1
 
-FROM --platform=linux/amd64 nvidia/cuda:$CUDA_VERSION-devel-centos7 AS cuda-build-amd64
+# Copy the minimal context we need to run the generate scripts
+FROM scratch AS llm-code
+COPY .git .git
+COPY .gitmodules .gitmodules
+COPY llm llm
```

Comment:
Note: I've found this change quite helpful in iterating when you're not touching the native code.  This refined/reduced context means that docker build doesn't have to re-generate if you're just mod'ing Go code outside of these paths making iterative `./scripts/build_linux.sh` much much faster.  If you do a git commit though it will taint the context, but I think that's inevitable since we need the git context to be able to run the generate script which needs to do submodule operations.